test/js/bun/glob/scan.test.ts

```diff
@@ -0,0 +1,321 @@
+const fgOpts = {
+  followSymbolicLinks: followSymlinks,
+  onlyFiles: false,
+  // absolute: true,
+} satisfies FgOpts;
+
+describe("glob.match", async () => {
+  function testWithOpts(namePrefix: string, bunGlobOpts: GlobScanOptions, fgOpts: FgOpts) {
+    test(`${namePrefix} recursively search node_modules`, async () => {
+      const pattern = "**/node_modules/**/*.js";
+      const glob = new Glob(pattern);
+      const filepaths = await Array.fromAsync(glob.scan(bunGlobOpts));
+      const fgFilepths = await fg.glob(pattern, fgOpts);
+
+      // console.error(filepaths);
+      expect(filepaths.length).toEqual(fgFilepths.length);
+
+      const bunfilepaths = new Set(filepaths);
+      for (const filepath of fgFilepths) {
+        if (!bunfilepaths.has(filepath)) console.error("Missing:", filepath);
+        expect(bunfilepaths.has(filepath)).toBeTrue();
+      }
+    });
+
+    test(`${namePrefix} recursive search js files`, async () => {
+      const pattern = "**/*.js";
+      const glob = new Glob(pattern);
+      const filepaths = await Array.fromAsync(glob.scan(bunGlobOpts));
+      const fgFilepths = await fg.glob(pattern, fgOpts);
+
+      expect(filepaths.length).toEqual(fgFilepths.length);
+
+      const bunfilepaths = new Set(filepaths);
+      for (const filepath of fgFilepths) {
+        if (!bunfilepaths.has(filepath)) console.error("Missing:", filepath);
+        expect(bunfilepaths.has(filepath)).toBeTrue();
+      }
+    });
+
+    test(`${namePrefix} recursive search ts files`, async () => {
+      const pattern = "**/*.ts";
+      const glob = new Glob(pattern);
+      const filepaths = await Array.fromAsync(glob.scan(bunGlobOpts));
+      const fgFilepths = await fg.glob(pattern, fgOpts);
+
+      expect(filepaths.length).toEqual(fgFilepths.length);
+
+      const bunfilepaths = new Set(filepaths);
+      for (const filepath of fgFilepths) {
+        if (!bunfilepaths.has(filepath)) console.error("Missing:", filepath);
+        expect(bunfilepaths.has(filepath)).toBeTrue();
+      }
+    });
+
+    test(`${namePrefix} glob not freed before matching done`, async () => {
+      const promise = (async () => {
+        const glob = new Glob("**/node_modules/**/*.js");
+        const result = Array.fromAsync(glob.scan(bunGlobOpts));
+        Bun.gc(true);
+        const result2 = await result;
+        return result2;
+      })();
+      Bun.gc(true);
+      const values = await promise;
+      Bun.gc(true);
+    });
+  }
+
+  testWithOpts("non-absolute", bunGlobOpts, fgOpts);
+  testWithOpts("absolute", { ...bunGlobOpts, absolute: true }, { ...fgOpts, absolute: true });
+
+  test("invalid surrogate pairs", async () => {
+    const pattern = `**/*.{md,\uD83D\uD800}`;
+    const cwd = "test/js/bun/glob";
+
+    const glob = new Glob(pattern);
+    const entries = await Array.fromAsync(glob.scan({ cwd }));
+
+    expect(entries.sort()).toEqual(
+      [
+        "fixtures/file.md",
```

Comment:
For the fixtures, instead of having them committed to source, I would recommend importing `createTmpDir` (not exact signature) from `"harness"`. This makes a bunch of temp files in a temp directory.

Reply:
Removed the fixtures from git also the tests need to scan the `test/js/bun/glob` dir so I made a function to create a temp `fixtures` dir there